spec/models/room_spec.rb

```diff
@@ -5,7 +5,31 @@
     expect(Room.new).not_to be_valid
     room = Room.new(name: "my room")
     room.choices.build(title: "trivial choice")
-    expect(room.save).to be
+    expect(room).to be_valid
+  end
+
+  context("slug attribute") do
+    it "parameterizes the name for the slug" do
+      room = Room.new(name: "hello world or poopin")
+      room.choices.build(title: "hello world")
+      room.slug_it_up
+      expect(room.slug).to eq("hello-world-or-poopin")
+    end
+
+    it "uses the id if name can't be parameterized" do
+      room = Room.new(name: "∂ƒ∆ƒ∆")
+      room.choices.build(title: "poopin")
+      room.slug_it_up
+      expect(room.slug).to eq(room.id.to_s)
+    end
+
+    it "uses parameterized name if at all possible" do
+      room = Room.new(name: "†his is delicious π")
+      room.choices.build(title: "Yes")
+      room.choices.build(title: "or yes")
+      room.slug_it_up
+      expect(room.slug).to eq("his-is-delicious")
```

Comment:
I would advocate that if it contains _any_ random characters it should use the id, otherwise it just looks like a completely different room name that has nothing to do with what the user wanted. But that's trivial, and it's fine how it is unless you want to change it.